### PR TITLE
break up hint and codemirror css

### LINKED ISSUE #3872: Editor Popover styles broken
Just noticed on master that the editor styles are broken for tab completion. 😬 

I'll fix it, that came from my migration of editor to styled-components. Looks like the vendored codemirror css didn't get loaded properly.

screen shot 2018-12-18 at 8 08 18 pm

### PR DESCRIPTION
**Pull request by rgbkrk**
This fixes #3872 while also cleaning up our rendering of both hints and codemirror areas themselves. I refactored a bit of the completion handling and included more comments to help the next developer in this corner of the code.

screen shot 2019-01-08 at 9 28 22 pm

screen shot 2019-01-08 at 9 29 54 pm

### CODE REVIEW

**Comment by rgbkrk** [comment]
My last commit(s) may break the build here, time to find out.

File: packages/editor/src/jupyter/complete.tsx [hunk 1]
```diff
@@ -18,54 +26,31 @@ export function formChangeObject(cm: CMI, change: EditorChange) {
...
-      span.className += "completion-type completion-type-" + completion.type;
-      span.setAttribute("title", completion.type);
-      elt.appendChild(span);
-      elt.appendChild(text);
-    }
-  }))
-});
+type CompletionMatch =
+  | string
+  | {
```
Comment (captainsafia) [inline packages/editor/src/jupyter/complete.tsx]: Nit: But I think this interface type should be defined under its own variable.

**Review by captainsafia — APPROVED** [review]
Thanks for tackling this! I left one small style nit that you can fix here or ignore for now.

**Review by stormpython — APPROVED** [review]
LGTM!

File: packages/editor/src/jupyter/complete.tsx (hunk 1 above)
Comment (rgbkrk) [reply packages/editor/src/jupyter/complete.tsx]: Addressed.

**Pull request merged**